tests/Unit/API/Site/Controllers/BaseController/BaseControllerTest.php

```
@@ -0,0 +1,319 @@
+							'type'    => 'string',
+							'default' => 'ProductDefault',
+						],
+						'variation' => [
+							'type'       => 'object',
+							'properties' => [
+								'id'         => [
+									'type'    => 'number',
+									'default' => 0,
+								],
+								'attributes' => [
+									'type'       => 'object',
+									'properties' => [
+										'name'  => [
+											'type' => 'string',
+										],
+										'color' => [
+											'type' => 'string',
+										],
+									],
+								],
+							],
+
+						],
+					],
+				],
+			],
+		];
+
+		$items = [
+			'country_code' => [
+				[
+					'product'   => 'ProductOne',
+					'variation' => [
+						'id'         => 1,
+						'attributes' => [
+							'name'  => 'TestAttribute',
+							'color' => 'red',
+						],
+					],
+				],
+				[
+					'product'          => 'ProductTwo',
+					'fieldNotInSchema' => 'fieldNotInSchema',
+					'variation'        => [ 'price' => 2 ],
+				],
+			],
+		];
+
+		$result   = $this->controller->get_matched_data_with_schema( $schema, $items );
+		$expected = [
+			'country_code' => [
+				[
+					'product'   => 'ProductOne',
+					'variation' => [
+						'id'         => 1,
+						'attributes' => [
+							'name'  => 'TestAttribute',
+							'color' => 'red',
+						],
+					],
+				],
+				[
+					'product'   => 'ProductTwo',
+					'variation' => [ 'id' => 0 ],
+				],
+			],
+		];
+		$this->assertEquals( $expected, $result );
+		$this->assertArrayNotHasKey( 'fieldNotInSchema', $result['country_code'][1] );
+
+	}
+
+	public function test_type_array_with_wrong_data() {
+		$schema = [
+			'country_code' => [
+				'type' => 'array',
+				'items' => [
+					'type' => 'string'
+				]
+			],
+			'rate'         => [
+				'type' => 'number',
+			],
+		];
+
+		$items = [
+			'country_code' => 'ES', // It is not array
+			'rate'         => 7,
+		];
+
+		//In case to receive an incorrect array, it will skip the schema matching and will continue to the next field.
```

Is this really the behaviour that we want to have? We define the schema so whoever is consuming the API can know what type of response to expect. But with this test we are basically saying if it doesn't match the schema then silently ignore. Which means whoever is consuming the API will get an unexpected response and possibly broken behaviour.

I understand you mentioned validating can be left for a followup PR, but it seems kind of counter intuitive to create a test for broken behaviour which we aren't handling yet.

Same for the next test with an object.